mithril-common/src/chain_observer/pallas_observer.rs

```diff
@@ -175,6 +181,58 @@ impl PallasChainObserver {
         Ok(utxo)
     }
 
+    /// Fetches the current stake distribution using the provided `statequery` client.
+    async fn do_stake_snapshots_state_query(
+        &self,
+        statequery: &mut Client,
+    ) -> StdResult<StakeSnapshot> {
+        statequery
+            .acquire(None)
+            .await
+            .map_err(|err| anyhow!(err))
+            .with_context(|| "PallasChainObserver failed to acquire statequery")?;
+
+        let era = queries_v16::get_current_era(statequery)
+            .await
+            .map_err(|err| anyhow!(err))
+            .with_context(|| "PallasChainObserver failed to get current era")?;
+
+        let state_snapshot = queries_v16::get_stake_snapshots(statequery, era, BTreeSet::new())
+            .await
+            .map_err(|err| anyhow!(err))
+            .with_context(|| "PallasChainObserver failed to get stake snapshot")?;
+
+        Ok(state_snapshot)
+    }
+
+    fn get_stake_pool_hash(&self, key: &Bytes) -> Result<String, ChainObserverError> {
+        let pool_hash = bech32::encode("pool", key.to_base32(), Variant::Bech32)
+            .map_err(|err| anyhow!(err))
+            .with_context(|| "PallasChainObserver failed to encode stake pool hash")?;
+
+        Ok(pool_hash)
+    }
+
+    async fn get_stake_distribution_snapshot(
```

Comment:
This function is not correctly named as it returns a stake distribution and not a stake distribution snapshot:
```suggestion
    async fn get_stake_distribution(
```